metals/src/main/scala/scala/meta/internal/metals/Diagnostics.scala

```diff
@@ -270,7 +270,8 @@ final class Diagnostics(
       // Scala 3 sets the diagnostic code to -1 for NoExplanation Messages. Ideally
       // this will change and we won't need this check in the future, but for now
       // let's not forward them.
-      if (d.getCode() != null && d.getCode() != "-1") ld.setCode(d.getCode())
+      if (d.getCode() != null && d.getCode().getLeft() != "-1")
```

Comment:
I am suprised it didn't fail with a warning :thinking: 

```suggestion
      if (d.getCode() != null && d.getCode().isLeft() && d.getCode().getLeft() != "-1")
```

Reply:
Yea for sure, I was expecting a "You're not comparing apples to oranges and this will likely never be true" message.